openapiGenerator.go

```diff
@@ -462,32 +621,109 @@ func (g *openapiGenerator) generateDescription(desc protomodel.CoreDesc) string
 	return strings.Join(strings.Fields(t), " ")
 }
 
+func (g *openapiGenerator) generateMultiLineDescription(desc protomodel.CoreDesc) string {
+	if !g.descriptionConfiguration.IncludeDescriptionInSchema {
+		return ""
+	}
+	comments, _ := g.parseComments(desc)
+	return comments
+}
+
+func (g *openapiGenerator) validationRules(desc protomodel.CoreDesc) []string {
+	_, validationRules := g.parseComments(desc)
+	return validationRules
+}
+
+func (g *openapiGenerator) parseComments(desc protomodel.CoreDesc) (comments string, validationRules []string) {
+	c := strings.TrimSpace(desc.Location().GetLeadingComments())
+	blocks := strings.Split(c, "\n\n")
+
+	var sb strings.Builder
+	for i, block := range blocks {
+		if shouldNotRenderDesc(strings.TrimSpace(block)) {
+			continue
+		}
+		if i > 0 {
+			sb.WriteString("\n\n")
+		}
+		var blockSb strings.Builder
+		lines := strings.Split(block, "\n")
+		for i, line := range lines {
+			if i > 0 {
+				blockSb.WriteString("\n")
+			}
+			l := strings.TrimSpace(line)
+			if shouldNotRenderDesc(l) {
+				continue
+			}
+			if strings.HasPrefix(l, validationMarker) {
```

Comment:
should we search for all `+kubebuilder` prefixes?

Reply:
Good catch, should fix